src/nativescript-angular/renderer.ts

```diff
@@ -29,8 +31,18 @@ export class NativeScriptRootRenderer extends RootRenderer {
 
 @Injectable()
 export class NativeScriptRenderer extends Renderer {
+    private componentProtoId: string;
+    private hasComponentStyles: boolean;
     constructor(private _rootRenderer: NativeScriptRootRenderer, private componentProto: RenderComponentType) {
         super();
+        let page =<Page>(<any>topmost()).currentPage;
```

Comment:
I propose we move the code that gets the current page to the root renderer and access it from there in child renderers. This will remove the duplication with `selectRootElement`